cli/generator-op-microservice/generators/microservice/templates/gql.service.ts

```diff
@@ -0,0 +1,88 @@
+import express from 'express';
+import { ApolloServer } from 'apollo-server-express';
+import fs from 'fs';
+import https from 'https';
+import http from 'http';
+import { mergeSchemas } from 'graphql-tools';
+const { ApolloLogExtension } = require('apollo-log');
+<% if (dbSupport) { -%>
+import mongoose from 'mongoose';
+<% } -%>
+
+import gqlSchema from './src/typedef.graphql';
+import { <%= resolverName %> as resolver } from './src/resolver';
+
+/* Setting port for the server */
+const port = process.env.PORT || 8080;
+
+const app = express();
+const extensions = [() => new ApolloLogExtension({
+  level: 'info',
+  timestamp: true,
+})];
+
+<% if (dbSupport) { -%>
+    /* Configuring Mongoose */
+    mongoose.plugin((schema: any) => { schema.options.usePushEach = true; });
+    mongoose.set('useNewUrlParser', true);
+    mongoose.set('useFindAndModify', false);
+    mongoose.set('useCreateIndex', true);
+    mongoose.set('useUnifiedTopology', true);
+
+    /* Establishing mongodb connection */
+    const dbCredentials = (process.env.DB_USER && process.env.DB_PASSWORD)
+      ? `${process.env.DB_USER}:${process.env.DB_PASSWORD}@`
+      : '';
+    const dbConnection = `mongodb://${dbCredentials}${process.env.DB_PATH}/${process.env.DB_NAME}`;
+
+    mongoose.connect(dbConnection, { useNewUrlParser: true, useCreateIndex: true }).catch(console.error);
+
+    mongoose.connection.on('error', error => {
+      console.error(error);
+    });
+<% } -%>
+
+/* Defining the Apollo Server */
+const apollo = new ApolloServer({
+  playground: process.env.NODE_ENV !== 'production',
+  schema: mergeSchemas({
+    schemas: [
+      gqlSchema,
+    ],
+    resolvers: [
+      resolver,
+    ],
+  }),
+  subscriptions: {
+    path: '/subscriptions',
+  },
+  formatError: error => ({
+    message: error.message,
+    locations: error.locations,
+    stack: error.stack ? error.stack.split('\n') : [],
+    path: error.path,
+  }),
+  extensions
+});
+
+/* Applying apollo middleware to express server */
+apollo.applyMiddleware({ app });
+
+/*  Creating the server based on the environment */
+const server = process.env.NODE_ENV !== 'test'
+  ? https.createServer(
+    {
+      key: fs.readFileSync(`./certs/server.key`),
```

Comment:
Wouldn't it be better to simply store the cert paths in an ENV Variable. @riginoommen I remember talking to you about this.

Reply:
Wouldn't it be better to simply store the cert paths in an ENV Variable. @riginoommen I remember talking to you about this.

Reply:
@diwanshi I suggested him to keep the certificates within the project directory, as that makes it easier to manage the certificates instead of using the global /etc/pki/... path.
And also this is a common practise used elsewhere.